test/Metadata/MetadataTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Db\Metadata;
+
+use Zend\Db\Adapter\Platform\Mysql as MysqlPlatform;
+use Zend\Db\Adapter\Platform\SqlServer as SqlServerPlatform;
+use Zend\Db\Adapter\Platform\Sqlite as SqlitePlatform;
+use Zend\Db\Adapter\Platform\Postgresql as PostgresqlPlatform;
+use Zend\Db\Adapter\Platform\Oracle as OraclePlatform;
+use Zend\Db\Metadata\Metadata;
+
+class MetadataTest extends \PHPUnit_Framework_TestCase
+{
+    public function testTheConstructorSetsTheAdaptersAndTheSource()
+    {
+        $platformMock = new MysqlPlatform();
+
+        /** @var \Zend\Db\Adapter\Adapter $adapterMock */
+        $adapterMock = $this->getMockBuilder('Zend\Db\Adapter\Adapter')
+                            ->disableOriginalConstructor()
+                            ->getMock();
+
+        $adapterMock->expects($this->once())
+                    ->method('getPlatform')
+                    ->willReturn($platformMock);
+
+        $metadata = new Metadata($adapterMock);
+
+        $adapterReflectionProperty = new \ReflectionProperty($metadata, 'adapter');
+        $adapterReflectionProperty->setAccessible(true);
+        $adapterPropertyValue = $adapterReflectionProperty->getValue($metadata);
+
+        $sourceReflectionProperty = new \ReflectionProperty($metadata, 'source');
+        $sourceReflectionProperty->setAccessible(true);
+        $sourcePropertyValue = $sourceReflectionProperty->getValue($metadata);
+
+        $this->assertSame($adapterMock, $adapterPropertyValue);
```

Comment:
Better if use `$this->assertAttributeSame()` and remove all that reflection logic